app/services/service_status.service.js

```diff
@@ -37,70 +47,110 @@ class ServiceStatusService {
     })
   }
 
-  /**
-   * Receives an array of statuses and returns it in the format required by the nunjucks template in the view.
-   */
-  static _mapArrayToStatusCells (rows) {
-    // Map each row in the array we've received
-    return rows.map(row => {
-      // A status row has only two elements:
-      // * The thing having its status reported, which is a standard text cell;
-      // * Its status, which is formatted numeric so that it's right justified on its row.
-      return [
-        { text: row[0] },
-        { text: row[1], format: 'numeric' }
-      ]
-    })
+  static async _getVirusScannerData () {
+    try {
+      const { stdout, stderr } = await exec('clamdscan --version')
+      return stderr ? `ERROR: ${stderr}` : stdout
+    } catch (error) {
+      return `ERROR: ${error.message}`
+    }
   }
 
-  static async _getImportData () {
-    return [
-      [
-        'Cell 1.1',
-        'Cell 1.2',
-        'Cell 1.3',
-        'Cell 1.4',
-        'Cell 1.5'
-      ],
-      [
-        'Cell 2.1',
-        'Cell 2.2',
-        'Cell 2.3',
-        'Cell 2.4',
-        'Cell 2.5'
-      ]
-    ]
+  static async _getRedisConnectivityData () {
+    try {
+      const { stdout, stderr } = await exec('redis-server --version')
+      return stderr ? `ERROR: ${stderr}` : stdout
+    } catch (error) {
+      return `ERROR: ${error.message}`
+    }
   }
 
-  static async _getVirusScannerData () {
-    return [
-      [
-        'Status',
-        'OK'
-      ]
-    ]
+  static async _getAddressFacadeData () {
+    const statusUrl = new URL('/address-service/hola', servicesConfig.addressFacade.url)
+    const result = await this._requestData(statusUrl)
+
+    return result.succeeded ? result.response.body : result.response
   }
 
-  static async _getCacheConnectivityData () {
-    return [
-      [
-        'Status',
-        'Connected'
-      ]
-    ]
+  static async _getChargingModuleData () {
+    const statusUrl = new URL('/status', servicesConfig.chargingModule.url)
+    const result = await this._requestData(statusUrl)
+
+    return result.succeeded ? result.response.headers['x-cma-docker-tag'] : result.response
   }
 
-  static async _getServiceVersionsData () {
-    return [
+  static async _requestData (url) {
+    // As of v12, the got dependency no longer supports CJS modules. This causes us a problem as we are locked into
+    // using these for the time being. Some workarounds are provided here: https://github.com/sindresorhus/got/issues/1789
+    // We have gone the route of using await import('got'). We cannot do this at the top level as Node doesn't support
+    // top level in CJS so we do it here instead.
+    const { got } = await import('got')
```

Comment:
At one point we were importing `got` within `go()` and passing it to each function so that it's only imported once. Happy to go along with this change just as long as we're sure there's no real performance impact from importing it every time `_requestData()` is used?

Reply:
It's still only imported once. Like `require()` any modules imported are cached. When another `import` call is made to the same module the ESM module loader will return the cached entry.

The same applies to something like the logger; we `require()` is in hundreds of files but Node is only loading it once. From then on it's returning a cached entry. It's also what makes trying to stub things in unit tests so much fun.

If it helps, we do the same thing in the [sroc-charging-module-api](https://github.com/DEFRA/sroc-charging-module-api/blob/main/app/services/charges/request_rules_service_charge.service.js#L123)